modules/standard/Filerator.chpl

```diff
@@ -210,29 +206,38 @@ module Chpl_rt_glob {
 }
 
 iter glob(pattern="*") {
-  var glb : Chpl_rt_glob.glob_t;
+  var glb : chpl_glob_c_interface.glob_t;
 
-  const err = Chpl_rt_glob.chpl_glob(pattern:c_string, 0, glb);
-  const num = Chpl_rt_glob.chpl_glob_num(glb);
-  if (num) then
-    for i in 0..num-1 do
-      yield Chpl_rt_glob.chpl_glob_index(glb, i): string;
+  const err = chpl_glob_c_interface.chpl_glob(pattern:c_string, 0, glb);
+  // TODO: Handle error cases better
+  if (err != 0 && err != chpl_glob_c_interface.GLOB_NOMATCH) then
+    __primitive("chpl_error", "unhandled error in glob()");
+  //
+  // Use safeCast here, and then back again, in order to avoid conditional
+  // in iterator in order to get better generated code, and to support
+  // 'num-1' without risk of overflow
+  //
+  const num = chpl_glob_c_interface.chpl_glob_num(glb).safeCast(int);
+  for i in 0..num-1 do
```

Comment:
I don't think the safe casting is terribly regrettable here and prefer it without the conditional.